Modules/_io/fileio.c

```diff
@@ -266,10 +270,13 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
     assert(PyFileIO_Check(state, self));
 #endif
     if (self->fd >= 0) {
+        PyMem_Free(self->stat_atopen);
+        self->stat_atopen = NULL;
```

Comment:
internal_close() already does the same, so it's redundant, no?

Reply:
Moderately, in the `self->closefd` is false case behavior would differ from _pyio, but that will be caught by the PyMem_Free + PyMem_New below, so think this is always redundant / not needed